### ISSUE TITLE
Would K-fac optimizer be worthwhile to add?

### ISSUE THREAD

**Issue by Joshuaalbert** [GPflow/GPflow#861]
I'm wondering if the Kronecker-Factored Approximate Curvature (K-FAC) optimizer (https://arxiv.org/abs/1503.05671) would be worthwhile to include as a gpflow.train.optimizer. It's a fast approximation to natural gradients, from my own experiments converges much faster than adam, and is included in tf already under `tf.contrib`, and soon just as part of tf.

**Comment by hughsalimbeni** [comment]
sounds like nice idea to me

**Comment by Joshuaalbert** [comment]
@hughsalimbeni Kfac is originally intended for neural networks, and would require a little extra work to make work for gpflow models, but that should mostly be book keeping work. What sort of qualities does the current natural gradient lack that we would want Kfac to have?

**Comment by hughsalimbeni** [comment]
The nat grad only works for the q_mu, q_sqrt parameters. I've heard mutterings of using [something like this](https://arxiv.org/pdf/1806.04854.pdf) for the remaining parameters, but I've not seen this done yet.

**Comment by markvdw** [comment]
The difficulty I see with KFAC like methods, is that they're designed for maximum likelihood objectives, which the ones we use in GPflow aren't. Same for the paper that Hugh links. @Joshuaalbert, do you have any specific ideas on how KFAC could be implemented for GPs? Would you just treat the variational objective as if it were a ML one?

**Comment by st--** [comment]
@Joshuaalbert it'd be great if you wanted to investigate this yourself - if it seems useful, please do open a pull request with your proposed changes - but I think this is beyond the scope of what the core gpflow maintainers can support!
